File: autojs/src/main/java/com/stardust/autojs/core/KeepAliveService/BootReceiver.java
package com.stardust.autojs.core.KeepAliveService;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

// 类名：BootReceiver（不变）
public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        // 监听手机开机完成广播
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            // 开机后自动启动当前APP
            Intent launchIntent = context.getPackageManager().getLaunchIntentForPackage(context.getPackageName());
            if (launchIntent != null) {
                launchIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                context.startActivity(launchIntent);
            }
        }
    }
}

File: app/src/main/java/org/autojs/autojs/receiver/BootReceiver.kt
package org.autojs.autojs.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Handler
import android.os.Looper
import org.autojs.autojs.Pref
import org.autojs.autojs.ui.splash.SplashActivity

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent?) {
        if (context == null || intent == null) return

        val action = intent.action
        if (
            action == Intent.ACTION_BOOT_COMPLETED
            || action == "android.intent.action.QUICKBOOT_POWERON"
            || action == Intent.ACTION_USER_PRESENT
            || action == "android.intent.action.MEDIA_MOUNTED"
            || action == Intent.ACTION_USER_UNLOCKED
        ) {
            if (Pref.isBootAutoStartEnabled()) {
                Handler(Looper.getMainLooper()).postDelayed({
                    val i = Intent(context, SplashActivity::class.java)
                    i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
                    i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
                    context.startActivity(i)
                }, 5000)
            }
        }
    }
}